snapcraft/internal/common.py

```diff
@@ -53,32 +54,35 @@ def assemble_env():
     return '\n'.join(['export ' + e for e in env])
 
 
-def run(cmd, **kwargs):
+def _run(cmd: List[str], caller, **kwargs):
```

Comment:
I think of the "caller" being "the thing that called this function." Perhaps this is a `runner`?

Reply:
this aligns well with our other implementations, good catch